provider/provider.go

```diff
@@ -785,3 +739,40 @@ func ensureMinimumBuildInputs(inputs resource.PropertyMap) bool {
 	}
 	return true
 }
+
+// canPreview returns true if inputs are resolved enough to perform a preview
+// build.
+func (p *dockerNativeProvider) canPreview(ctx context.Context, inputs resource.PropertyMap, urn resource.URN) (bool, error) {
+	var msg string
+	var returnWithoutBuild bool
+	// verify buildOnPreview is Known; if not, send warning and continue.
+	if inputs["buildOnPreview"].ContainsUnknowns() {
+		msg = "buildOnPreview is unresolved; cannot build on preview. Continuing without preview image build. " +
+			"To avoid this warning, set buildOnPreview explicitly, and ensure all inputs are resolved at preview."
+		returnWithoutBuild = true
```

Comment:
could we return early here? then we don't need to do the double verification on L#755. 
We could return `msg` instead of an error, and put the logging logic and error handling in the main function.

Reply:
Great idea, fixed!